client/db/bolt/db.go

```diff
@@ -437,12 +438,33 @@ func (db *BoltDB) UpdateOrder(m *dexdb.MetaOrder) error {
 	if len(md.Proof.DEXSig) == 0 {
 		return fmt.Errorf("cannot save order without DEX signature")
 	}
-	return db.ordersUpdate(func(master *bbolt.Bucket) error {
+	return db.ordersUpdate(func(ob, aob *bbolt.Bucket) error {
 		oid := ord.ID()
-		oBkt, err := master.CreateBucketIfNotExists(oid[:])
-		if err != nil {
-			return fmt.Errorf("order bucket error: %w", err)
+
+		// Create or move an order bucket based on order status. Orders
+		// that have status less than or equal to booked go in the
+		// ordersBucket. Finished orders go in the archivedOrdersBucket.
+		var oBkt *bbolt.Bucket
+		var err error
+		executed := md.Status > order.OrderStatusBooked
+		if executed {
+			if ob.Bucket(oid[:]) != nil {
+				ob.DeleteBucket(oid[:])
```

Comment:
Let's at least log the error from `DeleteBucket`.  But can you comment on the meaning if `ob.Bucket(oid[:])` _does_ return `nil`?  This would mean the order was either (1) never added to the active orders bucket, or (2) it is already in the archived orders bucket?

Reply:
Yes, either it wasn't added to the active orders, or was already deleted, so probably adding a second time. Going by the comment on `UpdateOrder` I thought that was probably expected. It just replaces whatever. Really delete/insert more than update.